### ISSUE TITLE
public func swipe(_ direction: SwipeResultDirection, force: Bool = false) not working when button tapped quickly multiple times.

### ISSUE THREAD

**Issue by keshavx11** [Yalantis/Koloda#395]
if !animationSemaphore.isAnimating

this condition is failing

**Comment by shaandsingh** [comment]
Not positive but isn't the idea that you have to wait for a swipe to finish animating before the method fires again?

**Comment by keshavx11** [comment]
@shaandsingh you are correct but from user's perspective, it still takes time even after the card has gone out of sight. For them the animation has completed and they can see only the new card. 
Looks like isAnimating takes more time to reset than it should.

Changing the drag speed from default to .fast made the experience better though.

**Comment by shaandsingh** [comment]
Got it. Does swipe force = true help?

**Comment by keshavx11** [comment]
Just tried. Doesn't help.

**Comment by matthewjsloane** [comment]
+1 to this

**Comment by katiesmillie** [comment]
+ 1 as well, even with the .fast animation this issue still crops up

**Comment by hibernat** [comment]
+1 to this

**Comment by Mas0nSun** [comment]
+1 to this

**Comment by aya2453** [comment]
+1

**Comment by rogerluan** [comment]
+1

**Comment by silviyavelani** [comment]
+1

**Comment by andoma93** [comment]
+1

**Comment by anomaddev** [comment]
+1 still

**Comment by lidorf-gini-apps** [comment]
+ 1 still

**Comment by losikurya** [comment]
+ 1

**Comment by anomaddev** [comment]
Forked the repo and implemented a queue to semi fix this. The queue makes it to where the animation will run slightly delayed from the button press. In my use case the user is selecting one of three buttons based on what is displayed, so since the user won't be tapping the button till the next card is shown, this method works for me. If your use case is animating cards away as quickly as you can by tapping a button, this solution will probably not fix your issue.

[Forked Repo](https://github.com/anomaddev/Koloda.git)
